pkg/sql/stats/automatic_stats.go

```diff
@@ -265,11 +265,15 @@ func (r *Refresher) ensureAllTables(ctx context.Context, settings *settings.Valu
 		return
 	}
 
+	// We wait for the asOfTime delay and some more before initialization,
+	// to ensure that the historical read will pick up all system tables.
+	time.Sleep(time.Second)
+
 	rows, err := r.ex.Query(
 		ctx,
 		"get-tables",
 		nil, /* txn */
-		`SELECT table_id FROM crdb_internal.tables;`,
+		fmt.Sprintf(`SELECT table_id FROM crdb_internal.tables AS OF SYSTEM TIME '-%s'`, time.Microsecond),
```

Comment:
why 1 microsecond? Also I think sprintf is overkill given the value is hardcoded

Reply:
Added a comment to clarify. Good call on the constant, simplified.